Core/StitchCoreSDK/Tests/StitchCoreSDKTests/Auth/Internal/CoreStitchAuthUnitTests.swift

```diff
@@ -539,7 +900,79 @@ class CoreStitchAuthUnitTests: StitchXCTestCase {
         }
 
         XCTAssertTrue(auth.isLoggedIn)
-        XCTAssertNotNil(auth.authInfo)
+        XCTAssertNotNil(auth.activeUserAuthInfo)
         XCTAssertEqual(userToBeLinked.id, auth.user!.id)
+        XCTAssertEqual(auth.activeUserAuthInfo?.loggedInProviderType, StitchProviderType.userPassword)
+        XCTAssertEqual(auth.activeUserAuthInfo?.loggedInProviderName, UserPasswordAuthProvider.defaultName)
+    }
+
+    func testDeviceIdPersistance() throws {
+        let requestClient = getMockedRequestClient()
+        let routes = StitchAppRoutes.init(clientAppID: "my_app-12345").authRoutes
+        let storage = MemoryStorage.init()
+
+        let auth = try StitchAuth.init(
```

Comment:
[task] to truly test the persistence you should logout, create a new `auth` with the same storage, and log back in with the new `auth` to make sure that the device ID is the same as before. I think with the way the code is now, the device ID is persisted in memory between logins, but not between app restarts